src/Section/Data/Section.php

```diff
@@ -93,7 +95,7 @@ public static function rules(?ValidationContext $context = null): array
                 'string',
                 'max:255',
                 new HandleRule(['id', 'dateCreated', 'dateUpdated', 'uid', 'title']),
-                Rule::unique(Table::SECTIONS)->ignore($context->payload['sectionId'] ?? null),
+                Rule::unique(Table::SECTIONS)->ignore($context->payload['sectionId'] ?? null)->withoutTrashed('dateDeleted'),
```

Comment:
Added tests for this one and the others https://github.com/craftcms/cms/pull/18028/commits/1f4b462de594442fb0ab42469fe9340c7d390c74